### PR TITLE
Fix noInfringementStreak badges

### PR DESCRIPTION
**Pull request by Lunaliar**
# Description
There is a badge that awards users who go to certain lengths of months without acquiring or using a blue square. These badges were not being awarded despite the data and parameters being met. Testing this bug requires some patience as it takes a couple of minutes for badges to be updated on a user profile. You may need to switch between the dashboard and profile page as well as reload the app to see the new badge. You can also check if the badge is added by keeping an eye on the terminal log for the backend to see if “Added Badge” or “Replaced Badge” is logged. 

Fixes # (Badge Component #9 “no blue square for x months”)

## Mainly changes explained:

The bug originated from a misspelling with an incorrect camel case of a property of the user object.
I changed the variable from oldinfringements to oldInfringements and the function is working properly. 

## How to test:
check into current branch
do `npm install` and `...` to run this PR locally
log in as an admin user or any other user that can adjust start dates and blue squares
go to dashboard→ Profile→ Volunteer times→ Start Date
Remove all blue squares from the user
Change start date to simulate 3, 6, 12, 24, 36, and 48 months spans of time
Wait for the userProfile cronjob to run and award badges. Check for logged badge addition or replacement
Wait 5-10 min for the badge to update on the user. (May need to reload or switch pages)

## Note:
I noticed that even though the badges are added when the cronjob is run, it is not updated on the app until the second run or the cronjob sometimes even later.

### CODE REVIEW

**Comment by vitorsaa2k** [comment]
Hi @Lunaliar, I tested with all spans of time, and it all worked well. I'd like to add that you may forgoten about the 24 months badge and did't added it on the description, I tested it and it is working as well.

**Comment by Lunaliar** [comment]
> Hi @Lunaliar, I tested with all spans of time, and it all worked well. I'd like to add that you may forgoten about the 24 months badge and did't added it on the description, I tested it and it is working as well.

Hey you're absolutely right! I'll go ahead and update the PR, thanks!

**Comment by calexander3108** [comment]
Tested with 3, 6, 12, 24, 36, and 48 months spans of time. All seems to worked well, cronjob logging also worked (badge adding and badge replacing)

**Review by kangjay88 — APPROVED** [review]
Hi @Lunaliar, I have tested this PR with the time frames and it works as intended. Thank you!
PR292-1
PR292

**Comment by NyahJ** [comment]
Hi @Lunaliar，

I changed the start dates to simulate 3, 6, 12, 24, 36, and 48 months spans of time,
then the userProfile cronjob ran and awarded badges for both badge addition and replacement. 
Great job, thanks for your hard work!

截屏2023-03-18 上午11 39 00

**Comment by one-community** [comment]
Great job everyone, moving this to final review! There are many more badge bugs needing work if anyone who's reviewed this one is moving to the Dev team and would like to help with them.

**Comment by PrabhjotSembhi** [comment]
@Lunaliar Hey, Can you please resolve conflicts? So we do a final review and merge it

File: src/helpers/userHelper.js [hunk 1]
```diff
@@ -654,11 +833,31 @@ const userHelper = function () {
               }
             }
           } else if (user?.infringements?.length === 0) {
-            if (moment().diff(moment(user.createdDate), 'months', true) >= elem.months) {
-              if (user.oldinfringements.length === 0 || Math.abs(moment().diff(moment(user.oldinfringements[user.oldinfringements?.length - 1].date), 'months', true)) >= (elem.months - 12)) {
```
Comment (EvianTan) [inline src/helpers/userHelper.js:658]: Note for reviewers:
Here are the only functional/non-ESLint changes
oldinfrigements->oldInfrigements

**Review by EvianTan — APPROVED** [review]
Approved.

**Pull request merged**